### PR TITLE
HPCC-14785 Throw an error on attempt to remove super that's owned

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@AttilaVamos - please review

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -3043,7 +3043,23 @@ class CDistributedFileBase : public CInterface, implements INTERFACE
...
+                error.append(iter->query().queryProp("@name"));
+                if (!iter->next())
+                    break;
+                error.append(", ");
+            }
+            return true;
+        }
+        return false;
+    }
     bool canRemove(StringBuffer &reason,bool ignoresub=false)
```
Comment (AttilaVamos) [inline dali/base/dadfs.cpp]: A light comment: I think the name 'reason' of the StringBuffer parameter is a little bit misleading. I can read it 'the reason of can remove'. Maybe the 'result' or something similar would be better.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -5468,7 +5478,10 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
         assertex(conn.get()); // must be attached
         CriticalBlock block(sect);
         checkModify("CDistributedSuperFile::detach");
-        subfiles.kill();    
+        StringBuffer reason;
+        if (checkOwned(reason))
```
Comment (AttilaVamos) [inline dali/base/dadfs.cpp]: Same comment as above. I can read it 'the reason of the check Owned'. Maybe the 'result' or similar would be better.

**Comment by AttilaVamos** [comment]
@jakesmith two small comments, but it seems good.

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (jakesmith) [inline dali/base/dadfs.cpp]: Perhaps, but not new to this PR and used elsewhere in existing codes to similar effect.
So will leave as it is, can consider changing in a future tidyup.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**